utils/ansible/ftest/library/daos_hugepages.py

```diff
@@ -0,0 +1,157 @@
+    hugepages_size = module.params['size']
+
+    ### Huge Pages configuration
+    start_time = datetime.datetime.utcnow()
+    if not os.path.isfile(r"/sys/kernel/mm/transparent_hugepage/enabled"):
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg="Huge Pages not activated in kernel")
+
+    if not is_huge_pages_enabled():
+        try:
+            with open(r"/sys/kernel/mm/transparent_hugepage/enabled", 'w', encoding="utf8") as fd:
+                fd.write('madvise')
+        except Exception as ex:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg=f"Huge Pages could not be enabled: {ex}")
+        if not is_huge_pages_enabled():
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg="Huge Pages could not be enabled")
+
+    cp = subprocess.run([r'sysctl', r'vm.nr_hugepages'],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=3)
+    if cp.returncode != 0:
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg=f"Size of Huge Pages could not be read: {cp.stderr.decode('ascii')}")
+
+    hugepages_current_size = 0
+    stdout_str = cp.stdout.decode('ascii')
+    m = __HUGEPAGES_SYSCTL_PATTERN.match(stdout_str)
+    if m is None:
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg=f"Invalid size of huge pages from sysctl: {stdout_str}")
+    hugepages_current_size = int(m.groupdict()['size'])
+
+    if hugepages_size != hugepages_current_size:
+        if check_mode:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg=f"Invalid size of huge pages: {hugepages_current_size}")
+
+        cp = subprocess.run([r'sysctl', f"vm.nr_hugepages={hugepages_size}"],
+            stdout=subprocess.DEVNULL, stderr=subprocess.PIPE, timeout=3)
+        if cp.returncode != 0:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg=f"Size of Huge Pages could not be dynamically set: "
+                    "{cp.stderr.decode('ascii')}")
+
+        try:
+            with open(r"/etc/sysctl.d/50-hugepages.conf", "w", encoding="utf8") as fd:
+                fd.write(f"vm.nr_hugepages={hugepages_size}")
+        except Exception as ex:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg=f"Setup of Huge Pages size at boot could not be defined: "
+                    "{cp.stderr.decode('ascii')}")
+        cp = subprocess.run([r'sysctl', '-p'], stderr=subprocess.PIPE, timeout=3)
+        if cp.returncode != 0:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg=f"Setup of Huge Pages size at boot could not be applied: "
```

Comment:
(pylint-f-string-without-interpolation) Using an f-string that does not have any interpolated variables